protocol/proofs/chain-config-registry.md

```diff
@@ -0,0 +1,94 @@
+# Purpose
+
+Design a solution that makes it easier to add or update chain configurations for use in fault proof programs and
+separate the release of op-program from updates to chain configuration.
+
+# Problem Statement + Context
+
+Currently, op-program and kona embed the configurations for chains they support via the superchain-registry and these
+are committed to on-chain as part of the `FaultDisputeGame` absolute pre-state. For existing chains that activate hard
+forks as part of the superchain configuration, this works well and the new chain configuration is included in the same
+update to the absolute prestate that implements the new hard fork rules in the fault proof program.
+
+However, chains that activate hard forks after the initial superchain set or new chains deployed between hard forks,
+they are unable to use the same absolute prestate because it doesn't contain the latest chain configuration. A new
+op-program release is required but it then isn't governance approved and may have other dependencies (e.g a recent
+update to the version of go used means an update to MIPS.sol is required).
+
+The conflation of implementation and chain configuration also means that we need a different absolute prestate for
+devnet, testnet and mainnet - each typically only differing by the addition of the hard fork activation time.
+
+# Proposed Solution
+
+Updating chain configuration could be simplified if the implementation and chain configuration were separated rather
+than both being committed to as part of the same absolute pre-state hash. `op-program` and `kona` would no longer
+include the superchain-registry but instead always load the chain configuration by requesting it via a local key from
+the `PreimageOracle`.
+
+A new `ChainConfigRegistry` contract would be created which provides a mapping from L2 chain ID to a hash of the chain
+configuration. A permissioned method would be provided to update the chain configuration, restricted to the
+`ProxyAdminOwner`. The `ProxyAdminOwner` role is used here because changing the chain configuration affects the
+definition of the valid chain, similar to upgrading contracts.
```

Comment:
Following Isthmus, there will be a new `SuperchainConfig.upgrader()` role which will represent the global 'governance actor'. We should use that instead of the `ProxyAdminOwner`.